lib/src/json/setting_value.dart

```diff
@@ -17,18 +18,41 @@ class SettingsValue {
   @JsonKey(name: 'd')
   final double? doubleValue;
 
-  SettingsValue(
+  SettingValue(
       this.booleanValue, this.stringValue, this.intValue, this.doubleValue);
 
-  factory SettingsValue.fromJson(Map<String, dynamic> json) =>
-      _$SettingsValueFromJson(json);
+  factory SettingValue.fromJson(Map<String, dynamic> json) =>
+      _$SettingValueFromJson(json);
 
-  Map<String, dynamic> toJson() => _$SettingsValueToJson(this);
+  Map<String, dynamic> toJson() => _$SettingValueToJson(this);
+
+  bool equalsBasedOnSettingType(Object? other, SettingType settingType) {
+    if (identical(this, other)) {
+      return true;
+    }
+    if (other is SettingValue && runtimeType == other.runtimeType) {
+      if (settingType == SettingType.boolean) {
+        return booleanValue == other.booleanValue;
+      }
+      if (settingType == SettingType.string) {
+        return stringValue == other.stringValue;
+      }
+      if (settingType == SettingType.int) {
+        return intValue == other.intValue;
+      }
+      if (settingType == SettingType.double) {
+        return doubleValue == other.doubleValue;
+      }
+      throw ArgumentError(
+          "Setting is of an unsupported type (${settingType.name}).");
+    }
+    return false;
+  }
 
   @override
   bool operator ==(Object other) =>
```

Comment:
Is this still used somewhere? I'd like it to be removed if possible, together with `hashCode`, because we don't want value equality semantics in the case of config model classes. Equality of setting values can't be defined reliably without knowing the setting type anyway.

Reply:
True, I forgot to delete the old code. Thanks!